lua/lualine/components/filename.lua

```diff
@@ -32,8 +39,12 @@ FileName.update_status = function(self)
 
   if data == '' then
     data = '[No Name]'
-  elseif vim.fn.winwidth(0) <= 84 or #data > 40 then
-    data = vim.fn.pathshorten(data)
+  end
+
+  for i = 0, count(data, "/"), 1 do
```

Comment:
You're hard coding path separator but remember windos uses forward slash instead of back slash .You can use `package.config:sub(1,1)` as path separator if I remember right . See the branch component for example .